lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -288,30 +256,171 @@ def create_instance(instance_params):
+        tacp_utils.wait_for_action_to_complete(
+            api_response.action_uuid, api_client)
+        result['ansible_module_results'] = tacp_utils.get_resource_by_uuid(
+            api_response.object_uuid, 'application', api_client)
+        result['changed'] = True
+
+    def instance_power_action(name, api_client, action):
+        instance_uuid = tacp_utils.get_component_fields_by_name(
+            module.params['name'], 'application', api_client)
+        api_instance = tacp.ApplicationsApi(api_client)
+
+        actions = {"start": api_instance.start_application_using_put,
+                   "stop": api_instance.stop_application_using_put,
+                   "restart": api_instance.restart_application_using_put,
+                   "shutdown": api_instance.shutdown_application_using_put,
+                   "pause": api_instance.pause_application_using_put,
+                   "resume": api_instance.resume_application_using_put,
+                   "delete": api_instance.delete_application_using_delete}
+
+        try:
+            api_response = actions[action](
+                instance_uuid)
+            if module._verbosity >= 3:
+                result['api_response'] = str(api_response)
+        except ApiException as e:
+            response_dict = tacp_utils.api_response_to_dict(e)
+
+            if "Error" in response_dict['message']:
+                result['changed'] = False
+                result['failed'] = True
+                fail_with_reason(response_dict['message'])
+
+        success, failure_reason = tacp_utils.wait_for_action_to_complete(
+            api_response.action_uuid, api_client)
+        
+        if not success:
+            fail_with_reason(failure_reason)
+        result['changed'] = True
+
+    # Return the inputs for debugging purposes
+    result['args'] = module.params
+
+    # Define configuration
+    configuration = tacp.Configuration()
+    configuration.host = "https://manage.cp.lenovo.com"
+    configuration.api_key_prefix['Authorization'] = 'Bearer'
+    configuration.api_key['Authorization'] = module.params['api_key']
+    api_client = tacp.ApiClient(configuration)
+
+    instance_uuid = tacp_utils.get_component_fields_by_name(
+        module.params['name'], 'application', api_client)
+    if instance_uuid:
+        instance_is_present = True
+    else:
+        instance_is_present = False
 
-    result['create_instance_response'] = create_instance_response
+    if instance_is_present:
+        instance_properties = tacp_utils.get_resource_by_uuid(
+            instance_uuid, 'application', api_client)
+        current_state = instance_properties['status']
+        # 'Running', 'Shut down', 'Paused', 'Restarting'
 
-    # use whatever logic you need to determine whether or not this module
-    # made any modifications to your target
-    # if module.params['new']:
-    #     result['changed'] = True
+        if module.params['state'] == 'absent':
```

Comment:
You are repeating yourself a lot here. The easiest way to refactor is to create a state map. Basically, a dictionary that says, if the instance is in this state, and the requested state is this, then do this. You can either do this with a multi-dimensional dict, or a dict with tuples as keys would work too.